fog/sql_recovery_db/src/lib.rs

```diff
@@ -241,6 +242,57 @@ impl RecoveryDb for SqlRecoveryDb {
         Ok(maybe_index.map(|val| val as u64))
     }
 
+    fn get_ingress_key_records(
+        &self,
+        start_block_at_least: u64,
+    ) -> Result<Vec<IngressPublicKeyRecord>, Error> {
+        let conn = self.pool.get()?;
+
+        use schema::ingress_keys::dsl;
+        let key_records: Vec<models::IngressKey> = dsl::ingress_keys
+            .filter(dsl::start_block.ge(start_block_at_least as i64))
+            .load(&conn)?;
+
+        let mut key_records = key_records
+            .iter()
+            .map(
+                |ingress_key_data| -> Result<IngressPublicKeyRecord, Error> {
+                    let status = IngressPublicKeyStatus {
+                        start_block: key_records[0].start_block as u64,
+                        pubkey_expiry: key_records[0].pubkey_expiry as u64,
+                        retired: key_records[0].retired,
+                    };
+
+                    // Look in ingested_blocks table for last scanned block
```

Comment:
Its probably better to do this with a subquery to cut down on roundtrips to the SQL server (and not make this an N+1 problem). There's an example [here](https://github.com/mobilecoinfoundation/fog/blob/master/fog/sql_recovery_db/src/lib.rs#L300-L302)